java-client/session/src/main/java/io/deephaven/client/impl/AggSpecBuilder.java

```diff
@@ -174,13 +175,15 @@ public void visit(io.deephaven.api.agg.spec.AggSpecTDigest tDigest) {
 
     @Override
     public void visit(io.deephaven.api.agg.spec.AggSpecUnique unique) {
-        final AggSpecUnique.Builder builder = AggSpecUnique.newBuilder().setIncludeNulls(unique.includeNulls());
-        unique.nonUniqueSentinel().map(AggSpecBuilder::adapt).ifPresent(builder::setNonUniqueSentinel);
+        final AggSpecUnique.Builder builder = AggSpecUnique.newBuilder()
+                .setIncludeNulls(unique.includeNulls())
+                .setNonUniqueSentinel(adapt(unique.nonUniqueSentinel().orElse(null)));
         out = spec(Builder::setUnique, builder);
     }
 
-    private static AggSpecNonUniqueSentinel adapt(AnnotatedObject obj) {
-        return obj.visit(AggSpecNonUniqueSentinelAdapter.INSTANCE);
+    private static AggSpecNonUniqueSentinel adapt(UnionObject obj) {
+        return obj == null ? AggSpecNonUniqueSentinel.newBuilder().setNullValue(NullValue.NULL_VALUE).build()
```

Comment:
Well, it is "arbitrary" in the sense that we _could_ define this proto differently if we wanted to; but it is a well-known type/value as part of protobuf: https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#nullvalue. Programmatically, `NullValue.NULL_VALUE` is the only instance of type `google.protobuf.NullValue`.

For example, instead of:

```proto
  message AggSpecNonUniqueSentinel {
    oneof type {
      google.protobuf.NullValue null_value = 1;
```

we could do

```
  message AggSpecNonUniqueSentinel {
    oneof type {
      int32 null_value = 1;
```

and

```java
AggSpecNonUniqueSentinel.newBuilder().setNullValue(0).build()
```

Reply:
I'll give this a shot.